packages/react-instantsearch-core/src/connectors/connectDynamicWidgets.ts

```diff
@@ -30,8 +51,36 @@ export default createConnector({
         results.renderingContent.facetOrdering.facets.order) ||
       [];
 
+    const attributesToRender = props.transformItems(facetOrder, { results });
+
+    if (attributesToRender.length > MAX_WILDCARD_FACETS && !props.facets) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `More than ${MAX_WILDCARD_FACETS} facets are requested to be displayed without explicitly setting which facets to retrieve. This could have a performance impact. Set "facets" to [] to do two smaller network requests, or explicitly to ['*'] to avoid this warning.`
+      );
+    }
+
+    if (props.maxValuesPerFacet < results._state.maxValuesPerFacet) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `The maxValuesPerFacet set by dynamic widgets (${props.maxValuesPerFacet}) is smaller than one of the limits set by a widget (${results._state.maxValuesPerFacet}). This causes a mismatch in query parameters and thus an extra network request when that widget is mounted.`
```

Comment:
[same as for IS.js,](https://github.com/algolia/instantsearch.js/pull/4968/files) I think this message would be more helpful it mentioned the name of the widget or the attribute, and clearly said: your should set your maxValuesPerFacet to max(limit/showMoreLimit)

Reply:
I don't think we can know that, as the maxValuesPerFacet isn't related to a single widget, but to all of them